ipld/merkledag/merkledag.go

```diff
@@ -514,6 +538,21 @@ func parallelWalkDepth(ctx context.Context, getLinks GetLinks, root cid.Cid, vis
 						return
 					}
 
+					// Successfully fetched "ci".
+					// provide it when needed
+					if prov := options.Provider; prov != nil {
+						err := prov.Provide(ctx, ci, true)
+						if err != nil && options.ErrorHandler != nil {
+							err = options.ErrorHandler(root, err)
```

Comment:
💭  should error handling show the actual failing ci, or the root one?
```suggestion
							err = options.ErrorHandler(ci, err)
```

Reply:
Per Gui's suggestions, we are going to not abort operations on provide failures, so we are just going to log the failure (and it logs the CID in latest version of the code).